third-party/realdds/doc/DDS ICD.md

```diff
@@ -0,0 +1,147 @@
+
+There are 4 main topics, consisted of these 2 types.
+* **realdds/device-info** of type flexible - The server sends a sample of this topic to inform readers of a connected camera.
+  Some of the details included are the camera model and serial number, this information will be used in the other topic names to ease communication with verious cameras in the network.
+* **realdds/\<model\>/\<serial\>/notification** of type flexible - Notifications are sent from the camera server to the camera user.
+  Some notificaitons are being sent at discovery to convey the full device information, e.g streams, supported options, etc...
+  Other notifications are sent as needed, mostly as reply to a control message
+* **realdds/\<model\>/\<serial\>/control** of type flexible - Controls are commands from the camera user to the camera, e.g start streaming, set option value etc...
+* **realdds/\<model\>/\<serial\>/<stream name>** of type image - Frames of data of the appropriate stream from the camera server to the user.
+
+# Messages
+
+## device-info
+
+On the camera user side, `realdds::dds_device_watcher` can be used to listen to incoming device-info messages.
+When compiling librealsense with the `BUILD_WITH_DDS` flag the context automatically creates and uses a `realdds::dds_device_watcher` object to identify DDS connected cameras.
+
+The server side can use `realdds::dds_device_broadcaster` to add or remove cameras from the DDS network.
+When a camera is added to the network the broadcaster will crerate a dds writer for it. If there is a **realdds/device-info** reader that mathces the broadcaster writer, topic sample will be sent with the appropriate device details.
+When a camera is removed from the network the broadcaster will close the writer, thus informing possible readers that the camera is disconnected.
+
+### device-info message format example
+
+device-info uses the **flexible** type with a JSON format of the following structure
+
+    "locked":false,
+    "name":"Intel RealSense D405",
+    "product-id":"0B5B",
+    "product-line":"D400",
+    "serial":"123622270732",
+    "topic-root":"realsense/D405/123622270732"
+
+## Notifications
+
+Notifications are messages from the camera to the user.
+
+When a user receives a `device-info` topic he can create a reader for this camera notifications using the `topic-root` field of the message.
+This can also be achived by using `realdds::dds_device` that will create the reader and wait for discovery notification messages to build an appropriate `realdds::dds_stream`s.
+
+When using librealsense API, received notifications will automatically be handled and reflected in the `rs2::device` and `rs2::sensor`s state.
+
+All notifications use the **flexible** type with JSON format
+
+### Discovery notifications
+
+On the discovery phase the camera server will send a **device-header** message, a possible **device-options** message and for each stream a **stream-header** and **stream-options** messages.
```

Comment:
what is the meaning of "possible" device-options message?
Do you mean that it will sometimes send it and sometimes not? when will this happen or not?

Reply:
Most options are in the stream level - like gain, exposure and such. Not all cameras have options at device level, that is affecting all the streams. Will be sent if such an option is supported.

Added footnote to explain